app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -251,13 +248,6 @@ private static boolean shouldGenerateFirstKey() {
             e.printStackTrace();
             Logger.log(LogTypes.TYPE_WARNING_NETWORK, "Timed out listening to receive data during sync");
             responseError = PullTaskResult.CONNECTION_TIMEOUT;
-        } catch (ConnectTimeoutException e) {

```

Comment:
Is this no longer a valid exception for some reason? There's definitely still a difference between SocketTimeouts and Connection timeouts in retrofit, right? Do we need to add the new exception type for retrofit's verison?

Reply:
I removed it earlier since it was deprecated. and OkHTTP sends the same SocketTimeOutException in both cases currently. Though on looking at this https://github.com/square/okhttp/issues/3318 it seems like something like `ConnectTimeoutException` might get added in newer releases.